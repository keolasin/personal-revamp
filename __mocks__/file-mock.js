module.exports = {
	mobileImage: {
		childImageSharp: {
			fluid: {
				aspectRatio: 0.7987220447284346,
				src: "/static/b3bb68c33b441137e56536ad52f48d32/724c8/jTree.jpg",
				srcSet:
					"/static/b3bb68c33b441137e56536ad52f48d32/84d81/jTree.jpg 250w,\n/static/b3bb68c33b441137e56536ad52f48d32/f0719/jTree.jpg 500w,\n/static/b3bb68c33b441137e56536ad52f48d32/724c8/jTree.jpg 1000w,\n/static/b3bb68c33b441137e56536ad52f48d32/d79bd/jTree.jpg 1500w,\n/static/b3bb68c33b441137e56536ad52f48d32/a66ad/jTree.jpg 2000w,\n/static/b3bb68c33b441137e56536ad52f48d32/5bdee/jTree.jpg 3456w",
				srcSetWebp:
					"/static/b3bb68c33b441137e56536ad52f48d32/1d872/jTree.webp 250w,\n/static/b3bb68c33b441137e56536ad52f48d32/4e6d4/jTree.webp 500w,\n/static/b3bb68c33b441137e56536ad52f48d32/36ebb/jTree.webp 1000w,\n/static/b3bb68c33b441137e56536ad52f48d32/fd45d/jTree.webp 1500w,\n/static/b3bb68c33b441137e56536ad52f48d32/6e77b/jTree.webp 2000w,\n/static/b3bb68c33b441137e56536ad52f48d32/fa3d3/jTree.webp 3456w",
				sizes: "(max-width: 1000px) 100vw, 1000px",
				originalImg: "/static/b3bb68c33b441137e56536ad52f48d32/5bdee/jTree.jpg",
			},
		},
	},
	desktopImage: {
		childImageSharp: {
			fluid: {
				aspectRatio: 1.3966480446927374,
				src: "/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/724c8/hiking.jpg",
				srcSet:
					"/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/84d81/hiking.jpg 250w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/f0719/hiking.jpg 500w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/724c8/hiking.jpg 1000w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/d79bd/hiking.jpg 1500w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/a66ad/hiking.jpg 2000w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/a521d/hiking.jpg 4608w",
				srcSetWebp:
					"/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/1d872/hiking.webp 250w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/4e6d4/hiking.webp 500w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/36ebb/hiking.webp 1000w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/fd45d/hiking.webp 1500w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/6e77b/hiking.webp 2000w,\n/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/6bce6/hiking.webp 4608w",
				sizes: "(max-width: 1000px) 100vw, 1000px",
				originalImg:
					"/static/a4cd4b58bfdb8357ae84b4cb5b5c7a1a/a521d/hiking.jpg",
			},
		},
	},
};
