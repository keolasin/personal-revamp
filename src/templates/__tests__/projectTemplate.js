import React from "react";
import { render } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Project from "../projectTemplate";

// Mocks and fixate generate className
import { mobileImage, desktopImage } from "../../../__mocks__/file-mock.js";
jest.mock("short-uuid");

beforeEach(() => {
	useStaticQuery.mockImplementation(() => ({
		mobileImage,
		desktopImage,
	}));

	const uuid = require("short-uuid");
	uuid.generate.mockImplementation(() => "73WakrfVbNJBaAmhQtEeDv");
});

describe("Project", () => {
	// mock data
	const data = {
		project: {
			html: "<p>A chatroom app built with create-react-app and firebase.</p>",
			frontmatter: {
				title: "Chatter",
				date: "April 01, 2020",
				link: "https://bloc-chat-react-m1d0mp1ge.now.sh",
				imageAlt: "Riverwalk Talk",
			},
		},
	};

	it("renders correctly", () => {
		const { container } = render(<Project data={data} />);
		expect(container).toMatchSnapshot();
	});
});
