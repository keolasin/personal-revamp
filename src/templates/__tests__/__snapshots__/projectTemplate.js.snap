// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
.emotion-13 {
  width: 100%;
  min-height: 100vh;
  background-color: transparent;
}

.emotion-4 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  place-items: center center;
  width: 100%;
  min-height: 10vh;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-template-areas: ". home ." "photos projects contact";
  background-color: rgba(0,0,0,0.75);
  border-radius: 0px 0px 25px 25px;
}

@media (min-width:320px) {
  .emotion-4 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-4 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-4 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-4 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    font-size: 3rem;
  }
}

.emotion-4 a {
  color: #dbe7fb;
  -webkit-text-stroke-width: 1px;
  -webkit-text-stroke-color: black;
}

.emotion-4 a:hover {
  color: #798be4;
}

.emotion-4 a:focus {
  color: #bc9612;
}

.emotion-4 a:active {
  color: #bc9612;
}

@media (min-width:1200px) {
  .emotion-4 {
    width: 80%;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
    grid-template-areas: "home photos projects contact";
  }
}

.emotion-0 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  font-size: 1.8rem;
  grid-area: home;
  margin-top: 5px;
}

@media (min-width:320px) {
  .emotion-0 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-0 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 3rem;
  }
}

@media (min-width:320px) {
  .emotion-0 {
    font-size: 1.8rem;
    margin-top: 5px;
  }
}

@media (min-width:480px) {
  .emotion-0 {
    font-size: 2rem;
    margin-top: 10px;
    margin-bottom: 5px;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 2.3rem;
    margin-top: 10px;
    margin-bottom: 5px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    font-size: 2.5rem;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 3.5rem;
    margin-top: 20px;
    margin-bottom: 22px;
  }
}

.emotion-1 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  font-size: 1.4rem;
}

@media (min-width:320px) {
  .emotion-1 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-1 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-1 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-1 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    font-size: 3rem;
  }
}

@media (min-width:480px) {
  .emotion-1 {
    font-size: 1.6rem;
  }
}

@media (min-width:768px) {
  .emotion-1 {
    font-size: 1.8rem;
  }
}

@media (min-width:992px) {
  .emotion-1 {
    font-size: 2rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    font-size: 2.5rem;
  }
}

.emotion-12 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  overflow: auto;
}

@media (min-width:320px) {
  .emotion-12 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-12 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-12 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-12 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-12 {
    font-size: 3rem;
  }
}

@media (min-width:768px) {
  .emotion-12 {
    width: 95%;
  }
}

@media (min-width:992px) {
  .emotion-12 {
    width: 90%;
  }
}

@media (min-width:1200px) {
  .emotion-12 {
    width: 80%;
  }
}

.emotion-6 {
  border-radius: 25px;
  font-family: "astounder-squared-bb",sans-serif;
  color: #bc9612;
  font-size: 1.4rem;
  background-color: rgba(0,0,0,0.75);
  display: inline-block;
  padding: 0 20px;
  margin: 10px;
}

@media (min-width:480px) {
  .emotion-6 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    font-size: 2.3rem;
  }
}

.emotion-10 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  background-color: rgba(0,0,0,0.75);
  border-radius: 25px;
  text-align: left;
  color: #dbe7fb;
  font-family: "acumin-pro",sans-serif;
  line-height: 1.2em;
  font-size: 1rem;
  padding: 5px;
  margin: auto 0;
}

@media (min-width:320px) {
  .emotion-10 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-10 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-10 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-10 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    font-size: 3rem;
  }
}

@media (min-width:768px) {
  .emotion-10 {
    font-size: 1.2rem;
    margin: 10px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    font-size: 1.7rem;
  }
}

.emotion-8 {
  font-family: "astounder-squared-bb",sans-serif;
  font-size: 62.5%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  margin: 0 auto;
  color: #BC9612;
  font-weight: bold;
  font-size: 1rem;
  margin: auto 10px;
}

@media (min-width:320px) {
  .emotion-8 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-8 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .emotion-8 {
    font-size: 1.6rem;
  }
}

@media (min-width:992px) {
  .emotion-8 {
    font-size: 1.8rem;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    font-size: 3rem;
  }
}

@media (min-width:320px) {
  .emotion-8 {
    font-size: 1rem;
  }
}

@media (min-width:480px) {
  .emotion-8 {
    font-size: 1.1rem;
  }
}

@media (min-width:768px) {
  .emotion-8 {
    font-size: 1.2rem;
  }
}

@media (min-width:992px) {
  .emotion-8 {
    font-size: 1.3rem;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    font-size: 1.4rem;
  }
}

<div>
  <section
    aria-label="Art-directed fullscreen background image"
    class="emotion-13 emotion-14 gbi--1561385816-73WakrfVbNJBaAmhQtEeDv"
    role="img"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; background-color: transparent; position: relative; opacity: 0.99;"
    title="Fullscreen Background"
  >
    <style>
      
          .emotion-13.emotion-14.gbi--1561385816-73WakrfVbNJBaAmhQtEeDv:before,
          .emotion-13.emotion-14.gbi--1561385816-73WakrfVbNJBaAmhQtEeDv:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;
background-color: transparent;

          }
          .emotion-13.emotion-14.gbi--1561385816-73WakrfVbNJBaAmhQtEeDv:before {
            z-index: -100;
            
            
            opacity: 1; 
          }
          .emotion-13.emotion-14.gbi--1561385816-73WakrfVbNJBaAmhQtEeDv:after {
            z-index: -101;
            
            
            
          }
        
    </style>
    <noscript />
    <nav
      class="emotion-4 emotion-5"
    >
      <a
        class="emotion-0"
        href="/"
      >
        MReyes
      </a>
      <a
        class="emotion-1"
        href="/albums"
        style="grid-area: photos;"
      >
        Photos
      </a>
      <a
        class="emotion-1"
        href="/projects"
        style="grid-area: projects;"
      >
        Projects
      </a>
      <a
        class="emotion-1"
        href="/contact"
        style="grid-area: contact;"
      >
        Contact
      </a>
    </nav>
    <main
      class="emotion-12"
    >
      <h1
        class="emotion-6 emotion-7"
      >
        Chatter
      </h1>
      <article
        class="emotion-10 emotion-11"
      >
        <strong
          class="emotion-8 emotion-9"
        >
          April 01, 2020
        </strong>
        <section>
          <p>
            A chatroom app built with create-react-app and firebase.
          </p>
        </section>
      </article>
    </main>
  </section>
</div>
`;
